ddl/ddl_api.go

```diff
@@ -1051,6 +1063,26 @@ func checkDuplicateConstraint(namesMap map[string]bool, name string, foreign boo
 	return nil
 }
 
+func setEmptyCheckConstraintName(tableLowerName string, namesMap map[string]bool, constrs []*ast.Constraint) {
+	cnt := 1
+	constraintPrefix := tableLowerName + "_chk_"
+	for _, constr := range constrs {
+		if constr.Name == "" {
+			constrName := fmt.Sprintf("%s%d", constraintPrefix, cnt)
+			for {
+				// loop until find constrName that haven't been used.
+				if !namesMap[constrName] {
+					namesMap[constrName] = true
+					break
+				}
+				cnt++
+				constrName = fmt.Sprintf("%s%d", constraintPrefix, cnt)
+			}
+			constr.Name = constrName
```

Comment:
move `namesMap[constrName] = true` to here is better. there may be bug if some one name its check `tableName_chk_1`, `tableName_chk_2`

Reply:
I think here is ok~
the passed parameter `namesMap` has record the used defined name already.

if some one name its check `tableName_chk_1`, `tableName_chk_2`, the loop will run for 3 time, and find `namesMap[3-times] = false` and break